kotlin/openID4VP/src/main/java/io/mosip/openID4VP/authorizationRequest/AuthorizationRequest.kt

```diff
@@ -72,24 +79,111 @@ data class AuthorizationRequest(
             } catch (exception: Exception) {
-                throw AuthorizationRequestExceptions.InvalidQueryParams("Exception occurred when extracting the query params from Authorization Request : ${exception.message}")
+                throw Logger.handleException(
+                    exceptionType = "InvalidQueryParams",
+                    message = "Exception occurred when extracting the query params from Authorization Request : ${exception.message}",
+                    className = className
+                )
             }
         }
 
+        private fun fetchPresentationDefinition(params: Map<String, String>): String {
+            val hasPresentationDefinition = params.containsKey("presentation_definition")
+            val hasPresentationDefinitionUri = params.containsKey("presentation_definition_uri")
+            var presentationDefinition = ""
+
+            when {
+                hasPresentationDefinition && hasPresentationDefinitionUri -> {
+                    throw Logger.handleException(
+                        exceptionType = "InvalidQueryParams",
+                        message = "Either presentation_definition or presentation_definition_uri request param can be provided but not both",
+                        className = className
+                    )
+                }
+
+                hasPresentationDefinition -> {
+                    val value = params["presentation_definition"] ?: throw Logger.handleException(
+                        exceptionType = "MissingInput",
+                        fieldPath = listOf("presentation_definition"),
+                        className = className
+                    )
+                    require(isNeitherNullNorEmpty(value)) {
+                        throw Logger.handleException(
+                            exceptionType = "InvalidInput",
+                            fieldPath = listOf("presentation_definition"),
+                            className = className
+                        )
+                    }
+                    presentationDefinition =
+                        params["presentation_definition"]!!
+                }
+
+                hasPresentationDefinitionUri -> {
+                    try {
+                        val value =
+                            params["presentation_definition_uri"] ?: throw Logger.handleException(
+                                exceptionType = "MissingInput",
+                                fieldPath = listOf("presentation_definition_uri"),
+                                className = className
+                            )
+                        require(isNeitherNullNorEmpty(value)) {
+                            throw Logger.handleException(
+                                exceptionType = "InvalidInput",
+                                fieldPath = listOf("presentation_definition_uri"),
+                                className = className
+                            )
+                        }
```

Comment:
All these scenarios are for Invalid input
param["presentation_definition_uri"] = null 
param["presentation_definition_uri"] = "null"
param["presentation_definition_uri"] = ""

Reply:
Handled it and first scenario is invalid because when sending the presentation_definition_uri as part of the request by default it's value will be considered as string so if we again pass the value wrapped in double quotes as presentation_definition_uri="null" then library will considered "null" as the actual value instead of taking it as null. So for this scenario it will go ahead with the other validations.